rice-framework/krad-web-framework/src/main/java/org/kuali/rice/krad/web/filter/UserLoginFilter.java

```diff
@@ -147,11 +148,22 @@ private boolean isAuthorizedToLogin(String principalId) {
      * Creates a session id cookie if one does not exists.  Write the cookie out to the response with that session id.
      * Also, sets the cookie on the established user session.
      */
-    private void establishSessionCookie(HttpServletRequest request, HttpServletResponse response) {
+    protected void establishSessionCookie(HttpServletRequest request, HttpServletResponse response) {
         String kualiSessionId = this.getKualiSessionId(request.getCookies());
         if (kualiSessionId == null) {
             kualiSessionId = UUID.randomUUID().toString();
-            response.addCookie(new Cookie(KRADConstants.KUALI_SESSION_ID, kualiSessionId));
+            Cookie sessionCookie = new Cookie(KRADConstants.KUALI_SESSION_ID, kualiSessionId);
+
+            // Make cookie secure
+            sessionCookie.setSecure(true);
+
+            // Make cookie HttpOnly if supported by container
+            Method setHttpOnlyMethod = ReflectionUtils.findMethod(Cookie.class, "setHttpOnly", boolean.class);

```

Comment:
Can you explain why ReflectionUtils is needed to determine if the container supports HttpOnly? Won't the `setHttpOnly` method always be present regardless?